libcomposefs/lcfs-writer-erofs.c

```diff
@@ -1116,12 +1116,20 @@ static int add_overlayfs_xattrs(struct lcfs_node_s *node)
 					  "", 0);
 		if (ret < 0)
 			return ret;
+		ret = lcfs_node_set_xattr(node, OVERLAY_XATTR_USERXATTR_WHITEOUT,
```

Comment:
Hum, interesting...but can you help me understand this?  Do we need the userxattrs for unprivileged overlayfs mounts?  This seems tricky because it's redundant data in the privileged case, no?  Or maybe it isn't much data and doesn't matter.

Reply:
So, a traditional whiteout is a chardev(0,0), and having that in the lowerdir will work as a whiteout for both regular and userxattr overlayfs mounts. Unfortunately they don't work in composefs, because the outer overlayfs eats them.

So, we have a new type of whiteout. If we set an escaped whiteout xattr like `trusted.overlay.overlay.whiteout` on a file in the erofs, then in the overlayfs mount it will show up as `trusted.overlay.whiteout`, and a (new) regular overlayfs mount will treat it as a whiteout. However, to a userxattr overlayfs mount it will not be special in any way, because it looks for `user.overlay.whiteout` files. So, composefs needs to set both of these for the resulting file to always work as a whiteout.

In terms of redundancy. Its not ideal, but it is what we need, and I don't think we'll have tons of whiteouts in practice.